codecov_auth/authentication/repo_auth.py

```diff
@@ -253,86 +256,64 @@ def authenticate_credentials(self, token):
 
 
 class TokenlessAuthentication(authentication.TokenAuthentication):
-    """This is named ironically, but provides authentication for tokenless uploads.
-    Tokenless only works in FORKs of PUBLIC repos.
-    It allows PRs from external contributors (forks) to upload coverage
-    for the upstream repo when running in a PR.
-
-    While it uses the same "shell" (authentication.TokenAuthentication)
-    it doesn't really rely on tokens to authenticate.
-    """
-
-    auth_failed_message = "Not valid tokenless upload"
-    rate_limit_failed_message = "Tokenless has reached GitHub rate limit. Please upload using a token: https://docs.codecov.com/docs/adding-the-codecov-token."
-
-    def _get_repo_info_from_request_path(self, request) -> Repository:
+    def _get_info_from_request_path(
+        self, request
+    ) -> tuple[Repository, str | None] | None:
         path_info = request.get_full_path_info()
         # The repo part comes from https://stackoverflow.com/a/22312124
-        upload_views_prefix_regex = r"\/upload\/(\w+)\/([\w\.@:_/\-~]+)\/commits"
+        upload_views_prefix_regex = (
+            r"\/upload\/(\w+)\/([\w\.@:_/\-~]+)\/commits(?:\/([a-f0-9]{40}))?"
```

Comment:
did this regex change?

Reply:
yes, there's an optional commit sha group at the end now